econml/dml.py

```diff
@@ -11,14 +11,42 @@
 
 import numpy as np
 import copy
-from .utilities import shape, reshape, ndim, hstack, cross_product, transpose
+from .utilities import (shape, reshape, ndim, hstack, cross_product, transpose,
+                        broadcast_unit_treatments, reshape_treatmentwise_effects, StatsModelsWrapper)
 from sklearn.model_selection import KFold, StratifiedKFold
 from sklearn.linear_model import LinearRegression, LassoCV
 from sklearn.preprocessing import PolynomialFeatures, LabelEncoder, OneHotEncoder
 from sklearn.base import clone, TransformerMixin
 from sklearn.pipeline import Pipeline
 from sklearn.utils import check_random_state
 from .cate_estimator import LinearCateEstimator
+from .inference import StatsModelsInference
+
+
+class _EffectOperation:
```

Comment:
This class needs a lot more commenting to explain what logic it abstracts and where it is potentially being used and how. This is a quite complex part of the code.

Reply:
I agree; however this factoring of the logic is internal to the library and not something that external users will ever need to see.  It needs to be documented anyway, so that our own contributors can make sense of it, but I'm hoping that we'll eventually come up with a better abstraction that we can use in place of this one because it doesn't seem natural.

Reply:
This class has been removed.